### PR TITLE
feat(quic): allow setting the MTU discovery upper bound

### PR DESCRIPTION
**Pull request by Wiezzel**
## Description

The default MTU discovery upper bound is too high for some environments and results in permanent warnings. As I'd like to be able to configure this setting, adding a new method for the `Config`.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by Wiezzel** [comment]
@jxs

File: transports/quic/src/config.rs [hunk 1]
```diff
@@ -90,6 +90,12 @@ impl Config {
         }
     }
 
+    // Set MTU discovery config
+    pub fn with_mtu_discovery_config(mut self, config: MtuDiscoveryConfig) -> Self {
```
Comment (jxs) [inline transports/quic/src/config.rs]: we should not expose `quinn`'s internal types. 
Meanwhile I checked and we already expose disabling MTU discovery: https://github.com/libp2p/rust-libp2p/. We can define a  `set_mtu_upper_bound(value: u16)`  wdyt?

**Review by jxs — COMMENTED** [review]
Thanks Adam, left a comment

File: transports/quic/src/config.rs (hunk 1 above)
Comment (Wiezzel) [reply transports/quic/src/config.rs]: I don't see an issue with re-exporting types from other libraries, but `set_mtu_upper_bound(value: u16)` works for me. Let me change it.

Comment (mxinden) [reply transports/quic/src/config.rs]: > I don't see an issue with re-exporting types from other libraries

Exposing types on other libraries, forces libp2p to make a breaking change, whenever the upstream library makes a breaking change, whether it is an actual breaking change for libp2p or not.

Does that make sense @Wiezzel?

Comment (Wiezzel) [reply transports/quic/src/config.rs]: >Exposing types on other libraries, forces libp2p to make a breaking change, whenever the upstream library makes a breaking change, whether it is an actual breaking change for libp2p or not.

I could argue that when `quinn` adds or removes a an option from `MtuDiscoveryConfig`, and libp2p maintainer makes a decision to include this change, it should probably be made available to libp2p users as well. But that's definitely not a hill I want to die on, so I've modified the code to expose just `with_mtu_upper_bound`.

File: transports/quic/src/config.rs [hunk 2]
```diff
@@ -90,6 +90,12 @@ impl Config {
         }
     }
 
+    // Set MTU discovery config
+    pub fn with_mtu_upper_bound(mut self, value: u16) -> Self {
```
Comment (jxs) [inline transports/quic/src/config.rs]:
```suggestion
    /// Set the upper bound to the max UDP payload size that MTU discovery will search for.
    pub fn mtu_upper_bound(mut self, value: u16) -> Self {
```

**Review by jxs — APPROVED** [review]
Thanks!

**Pull request merged**